### PR TITLE
Propagate environment C and C++ flags

### PR DESCRIPTION
**Pull request by okias**
Debian and Mobian would be very grateful, if build-system would be able to accept distribution options (in case of Debian mostly focus on hardening) by default without need of patching.

Followup of https://github.com/minetest/minetest/pull/14941

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -850,7 +851,7 @@ else()
 		endif()
 	endif()
 
-	set(CMAKE_CXX_FLAGS_RELEASE "-DNDEBUG ${RELEASE_WARNING_FLAGS} ${OTHER_FLAGS} -pipe -funroll-loops -O3 -fomit-frame-pointer")
+set(CMAKE_CXX_FLAGS_RELEASE "-DNDEBUG ${RELEASE_WARNING_FLAGS} ${OTHER_FLAGS} -pipe -funroll-loops -O3 -fomit-frame-pointer $ENV{CPPFLAGS}")
```
Comment (sfan5) [inline src/CMakeLists.txt:854]:
```suggestion
	set(CMAKE_CXX_FLAGS_RELEASE "-DNDEBUG ${RELEASE_WARNING_FLAGS} ${OTHER_FLAGS} -pipe -funroll-loops -O3 -fomit-frame-pointer $ENV{CPPFLAGS}")
```

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -37,6 +37,7 @@ if(NOT (CMAKE_BUILD_TYPE IN_LIST SUPPORTED_BUILD_TYPES))
 			"Unknown CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}. Options are: ${SUPPORTED_BUILD_TYPES}.")
 endif()
 
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} $ENV{CFLAGS} $ENV{CPPFLAGS}")
```
Comment (sfan5) [inline src/CMakeLists.txt:40]: but CPPFLAGS is for C++ only?

Comment (okias) [reply src/CMakeLists.txt]: I think this was initial mistake which got into final patch, I removed CPPFLAGS.

**Review by nerzhul — APPROVED** [review]

**Review by sfence — APPROVED** [review]

**Review by grorp — APPROVED** [review]

**Review by SmallJoker — APPROVED** [review]

**Comment by Desour** [comment]
Aren't CPPFLAGS the c preprocessor flags?
CFLAGS and CXXFLAGS are already used by cmake as default for `CMAKE_C_FLAGS` and `CMAKE_CXX_FLAGS` on my machine. (Haven't tried if it also already works for CPPFLAGS.)
(Distros might use CPPFLAGS to apply flags to both C and C++, I guess.)

**Comment by appgurueu** [comment]
Relevant documentation: https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_FLAGS.html

With that in mind the changes here seem a bit hacky:

* Why are changes needed for `CFLAGS` to work at all?
* If we want `CPPFLAGS` to work, we should stuff it in `CMAKE_CXX_FLAGS`, shouldn't we? That way it should also automatically be applied irrespective of build type.
* Related question: Why is it that in Minetest's CMakeLists, only the release flags have CPPFLAGS applied?

We can resume the approval chain after that is resolved :smile:

**Pull request closed without merging**